### PR TITLE
New DDS profile for a flaky test

### PR DESCRIPTION
**Pull request by tcappellari-bdai**
## Change Overview

Added in a new DDS profile based off of https://github.com/bdaiinstitute/bdai/pull/6448 to fix a flaky test caused by race conditions in service discovery in FastDDS.

## Testing Done

[x] I ran the pytests locally a few times and saw the flaky test pass but unsure how to robustly test it since it usually passes in CI but not always

Please create a checklist of tests you plan to do and check off the ones that have been completed successfully. Ensure that ROS 2 tests use `domain_coordinator` to prevent port conflicts. Further guidance for testing can be found on the [ros utilities wiki](https://github.com/bdaiinstitute/ros_utilities/wiki/Testing-guidelines).

### CODE REVIEW

**Review by khughes-bdai — APPROVED** [review]
LGTM as long as CI passes

File: spot_driver/test/CMakeLists.txt [hunk 1]
```diff
@@ -130,6 +130,8 @@ target_include_directories(test_kinematic_service
 )
 target_link_libraries(test_kinematic_service spot_api)
 
+ament_add_pytest_test(spot_driver_pytest ${CMAKE_CURRENT_SOURCE_DIR} TIMEOUT 900)
```
Comment (amessing-bdai) [inline spot_driver/test/CMakeLists.txt:133]: Do we want to go with this approach of 900s for all of the tests or the one below which loops over and gives individual tests targets?

Comment (amessing-bdai) [reply spot_driver/test/CMakeLists.txt:133]: To be clear I am not leaning one way or the other. Just checking

Comment (tcappellari-bdai) [reply spot_driver/test/CMakeLists.txt:133]: i think just this one since it seems to be the only (relatively) consistent flaky test

Comment (amessing-bdai) [reply spot_driver/test/CMakeLists.txt:133]: I mean do we want one target for all python tests or one target for each python test. The former has us set a timeout for all tests. The later lets use define a timeout for each test file.

Comment (tcappellari-bdai) [reply spot_driver/test/CMakeLists.txt:133]: ohhhh, ok i misunderstood 
I think one target for all tests is fine

**Pull request merged**